src/lib.rs

```diff
@@ -810,6 +810,61 @@ impl fmt::Debug for Connection {
     }
 }
 
+/// Batch iterator
+/// ```rust
+/// use fallible_iterator::FallibleIterator;
+/// use rusqlite::{Batch, Connection, Result, NO_PARAMS};
+///
+/// fn main() -> Result<()> {
+///     let conn = Connection::open_in_memory()?;
+///     let sql = r"
+///     CREATE TABLE tbl1 (col);
+///     CREATE TABLE tbl2 (col);
+///     ";
+///     let mut batch = Batch::new(&conn, sql);
+///     while let Some(mut stmt) = batch.next()? {
+///         stmt.execute(NO_PARAMS)?;
+///     }
+///     Ok(())
+/// }
+/// ```
+#[derive(Debug)]
+pub struct Batch<'conn, 'sql> {
+    conn: &'conn Connection,
+    sql: &'sql str,
+    tail: usize,
+}
+
+impl<'conn, 'sql> Batch<'conn, 'sql> {
+    /// Constructor
+    pub fn new(conn: &'conn Connection, sql: &'sql str) -> Batch<'conn, 'sql> {
+        Batch { conn, sql, tail: 0 }
+    }
+}
+
+impl<'conn> fallible_iterator::FallibleIterator for Batch<'conn, '_> {
```

Comment:
Is there a reason not to `impl Iterator<Item=Result<Statement<'conn>, Error>>`? I'm less sold on the benefit of FallibleIterator now that `transpose()` is stable and can trivially change between `Result<Option<T>, E>` and `Option<Result<T, E>>`.

Implementing both wouldn't bother me if you prefer the FallibleIterator API, but IMO not having to bring in the fallible iterator crate and import the trait to use this type is nicer for most users.

Reply:
I guess we can use the [iterator](https://docs.rs/fallible-iterator/0.2.0/fallible_iterator/trait.FallibleIterator.html#method.iterator) method.

Reply:
We should not implements both!

Reply:
Should be ok now: you can use either `while let Some(stmt) = batch {...}` or `for stmt in batch {...}`.